sdk/parents/azure-client-sdk-parent/pom.xml

```diff
@@ -110,46 +110,26 @@
   </properties>
 
   <build>
+    <resources>
+      <resource>
+        <directory>src/main/resources</directory>
+        <filtering>true</filtering>
+      </resource>
+    </resources>
+
     <plugins>
+      <!-- This plugin configures the bas requirements for compilation -->
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
         <version>3.8.1</version> <!-- {x-version-update;org.apache.maven.plugins:maven-compiler-plugin;external_dependency} -->
-        <configuration>
-          <showWarnings>true</showWarnings>
-          <failOnWarning>true</failOnWarning>
-        </configuration>
       </plugin>
 
       <!-- This plugin scans checkstyle issues in the code -->
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-checkstyle-plugin</artifactId>
         <version>3.1.1</version> <!-- {x-version-update;org.apache.maven.plugins:maven-checkstyle-plugin;external_dependency} -->
-        <dependencies>
```

Comment:
There's a `<reporting>` section that can probably be cleaned up as well. Also, the javadoc section with all the groups can be removed too.

Reply:
Removed all aggregate reporting configurations, leaving the non-aggregate configurations for now as children POMs inherit it. We can have a follow-up investigation on whether those should be removed as well.